lib/katello/tasks/reimport.rake

```diff
@@ -16,6 +16,22 @@ namespace :katello do
     User.current = User.anonymous_admin #set a user for orchestration
     Dir.glob(Katello::Engine.root.to_s + '/app/models/katello/*.rb').each { |file| require file }
 
+    print "Fixing incorrect providers\n"
+    incorrect_provider_count = 0
+    error_msgs = []
+    Katello::Product.redhat.each do |product|
```

Comment:
I think the spirit of your PR is to fix this once and for all but this reimport task is more general purpose to fix out-of-sync things. What you likely want to do is add an upgrade rake task like this
 https://github.com/Katello/katello/commit/df86940f1ab1116b8b3126424ba6064aa69e4853#diff-adc47a3fb24f22bba9fa2f908cc1e0676804459186bd60f27dc94f1febbd1f06

That'd get run when the system is updated or if it gets hotfixed. Your the rest of your PR will prevent that script from needing to run again

Reply:
I agree, since the incorrect providers fix should only be a one time fix thanks to the changes you've made. You could just yank out the changes here and stick them in an upgrade task.

Reply:
The reimport task (without the new code) will fail if you've got products with incorrect providers, though. I have no problem moving it, but I'd probably want to direct users to that new rake task.

Reply:
By the time the user gets your new code here, the upgrade task should have already run. They shouldn't even need to run the rake task manually.  The installer I believe is responsible for running `foreman-rake upgrade:run`.

Reply:
ooh, I see what you're saying now!